### DISCUSSION TITLE
[Book 1, Chapter 9.1] Why reject vectors outside unit sphere?

### DISCUSSION THREAD

**Discussion by dimitry-ishenko** [RayTracing/raytracing.github.io#1369] [Q&A] [answered]
We generate vectors within unit cube and reject those that are outside of unit sphere. We then normalize them to be on the surface of the sphere.

My question is: why do we reject vectors outside unit sphere? They can be normalized just like the other ones, and will end up on the surface of the sphere.

What did I miss?

**Comment by rupsis** [comment]
Giving it a test, there isn't _much_ of a difference (Left: `(p.length_squared() < 1)`, Right: `(p.length_squared() < sqrt(2))`

![image](https://github.com/RayTracing/raytracing.github.io/assets/9646385/ef984c3e-8178-403d-8a7c-14a576f22dcc)

But considering this is a sampling method, increasing the sampling threshold (in this case, to the whole unit cube) will caused there to be a disproportionate a mount of samples caught in the corners. Thus causing a slight skew. 

That's my understanding at least. @hollasch can tell me if I'm wrong or not :)

**Reply by hollasch** [reply, marked as answer]
Yes, @rupsis is correct. All random points along the axes will lie inside the sphere. However, the cube diagonals will have not only that number of random points, but additional random points that were generated from the sphere to the unit corner — (√2)-1 more points, to be precise.

As a result, you get an image that looks like this:
![Non-uniform sphere distribution](http://lukas-polok.cz/proj/random_dir.png)

See [Generating random directions with uniform distribution
](http://lukas-polok.cz/tutorial_sphere.htm) for a good article that covers this.

**Reply by dimitry-ishenko** [reply]
@rupsis and @hollasch thanks guys, this makes sense now
